python/datasources/acs_health_insurance.py

```diff
@@ -0,0 +1,612 @@
+    # FileDiff = If the data has changed by diffing the old run vs the new run.
+    # (presumably to skip the write to bq step though not 100% sure as of writing this)
+    def upload_to_gcs(self, bucket):
+        male_state_params = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_MALE_SUFFIXES)
+
+        file_diff = url_file_to_gcs.url_file_to_gcs(
+            self.base_url, male_state_params, bucket,
+            self.get_filename(Sex.MALE, None, False))
+
+        female_state_params = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES)
+        file_diff = url_file_to_gcs.url_file_to_gcs(
+            self.base_url, female_state_params, bucket,
+            self.get_filename(Sex.FEMALE, None, False)) and file_diff
+
+        male_county_params = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_MALE_SUFFIXES, True)
+        file_diff = url_file_to_gcs.url_file_to_gcs(
+            self.base_url, male_county_params, bucket,
+            self.get_filename(Sex.MALE, None, True)) and file_diff
+
+        female_county_params = format_params(
+            HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES, True)
+        file_diff = url_file_to_gcs.url_file_to_gcs(
+            self.base_url, female_county_params, bucket,
+            self.get_filename(Sex.FEMALE, None, True)) and file_diff
+
+        # Iterates over the different race ACS variables,
+        # retrieves the race from the metadata merged dict
+        # writes the data to the GCS bucket and sees if file diff is changed
+        for prefix in HEALTH_INSURANCE_BY_RACE_GROUP_PREFIXES:
+            for prefix_key in prefix:
+                race_state_params = format_params(
+                    prefix, HEALTH_INSURANCE_BY_RACE_GROUP_SUFFIXES)
+                race = prefix[prefix_key][MetadataKey.RACE]
+                file_diff = url_file_to_gcs.url_file_to_gcs(
+                    self.base_url, race_state_params, bucket,
+                    self.get_filename(None, race, False)) and file_diff
+                race_county_params = format_params(
+                    prefix, HEALTH_INSURANCE_BY_RACE_GROUP_SUFFIXES, True)
+                file_diff = url_file_to_gcs.url_file_to_gcs(
+                    self.base_url, race_county_params, bucket,
+                    self.get_filename(None, race, True)) and file_diff
+
+        return file_diff
+
+    def write_to_bq(self, dataset, gcs_bucket):
+        # Get an ACS mapping of Fip Codes to State Names and county codes to county names
+        self.get_state_fips_mapping()
+        self.get_county_fips_mapping()
+
+        # Pull data from GCS and aggregate in memory
+        self.get_health_insurance_data_by_sex(
+            use_gcs=True, gcs_bucket=gcs_bucket)
+        self.get_health_insurance_data_by_race(
+            use_gcs=True, gcs_bucket=gcs_bucket)
```

Comment:
What's the reason that you don't also aggregate by age? Is that not possible with this data? I was imagining that 1-D aggregations by race, age, and sex would be doable

Reply:
I could aggregate by age, but the issue requirements at the end we would end up with 4 files, and this is how it was spelled out in the issue.  Let me know if this is needed.

Reply:
Just read the issue and it looks like it's because the data is organized as such (age/race and age/sex). I think the data we eventually want to show in the FE is broken down each by race, age, and sex (i.e. 1-D breakdowns), but having these tables is fine as that is computable from these.